core/src/main/java/de/danoeh/antennapod/core/storage/DBReader.java

```diff
@@ -898,6 +898,16 @@ public static NavDrawerData getNavDrawerData() {
         }
         final LongIntMap feedCounters = adapter.getFeedCounters(feedIds);
 
+        int feedFilter = UserPreferences.getFeedFilter();
+        if (feedFilter == UserPreferences.FEED_FILTER_COUNTER_ZERO) {
+            for (int i = 0; i < feeds.size(); i++) {
```

Comment:
Please turn the iteration order around (instead of `0...size` to `size...0`). The reason is that this sometimes skips indices when removing an item. You can therefore end up with feeds that have counter 0 and are still displayed.